core/src/main/java/org/apache/calcite/sql/fun/SqlTimestampAddFunction.java

```diff
@@ -93,11 +93,8 @@ public static RelDataType deduceType(RelDataTypeFactory typeFactory,
             MICROSECOND_PRECISION);
         break;
       default:
-        if (operandType2.getSqlTypeName() == SqlTypeName.TIME) {
-          type = typeFactory.createSqlType(SqlTypeName.TIME);
-        } else {
-          type = typeFactory.createSqlType(SqlTypeName.TIMESTAMP);
-        }
+        final SqlTypeName typeName = sanitize(operandType2.getSqlTypeName());
```

Comment:
We talked offline:

- Seems like the BQ implementation of TIMESTAMP_ADD uses SqlBasicFunction.create and doesn't go through this code path so we don't need to address this here.

Reply:
So I played around with MySql on db-fiddle and the reason why your results are always returning `DATETIME` is because your adding time units that are less than `1 DAY`.

So if I run `TIMESTAMPADD(DAY, 1, DATE '2022-12-12')` I would get the DATE `2022-12-13` and it maintains a `DATE` type

Reply:
as a general practice, let's not re-litigate previous functionality (in this case the MySQL TIMESTAMPADD function) when adding new functionality. If you think it's wrong, log a bug. But assume that the existing test is correct.

Reply:
Ya, that's what I was suggesting, but the BigQuery type derivation was also subtly wrong so I was just pointing out that we probably don't want to validate both dialects' functions using the same code since they have different type inference rules. I think this has been resolved offline; `SqlTimestampAddFunction` was not being used to validate the BQ function in the first place.